Dockerfile

```
@@ -0,0 +1,16 @@
+FROM python:3.12.3-alpine as base
+    RUN pip install poetry
+    WORKDIR /app
+    ENTRYPOINT ["poetry", "run", "flask", "run", "--host=0.0.0.0"]
+    COPY poetry.lock ./
+    COPY pyproject.toml ./
```

Just in case you're not aware, worth noting we can condense these to a single line:
```suggestion
    COPY poetry.lock pyproject.toml ./
```
It's a very minor optimisation in reducing the number of layers by 1, so I'd personally lean towards whatever you find easiest to read, but flagging in case it's new information